pkg/authorization/registry/role/virtual_registry.go

```diff
@@ -0,0 +1,148 @@
+package role
+
+import (
+	"fmt"
+	"strings"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	klabels "github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+
+	authorizationapi "github.com/openshift/origin/pkg/authorization/api"
+	policyregistry "github.com/openshift/origin/pkg/authorization/registry/policy"
+)
+
+type VirtualRegistry struct {
+	policyRegistry policyregistry.Registry
+}
+
+// NewVirtualRegistry creates a new REST for policies.
+func NewVirtualRegistry(policyRegistry policyregistry.Registry) Registry {
+	return &VirtualRegistry{policyRegistry}
+}
+
+func (m *VirtualRegistry) ListRoles(ctx kapi.Context, labels, fields klabels.Selector) (*authorizationapi.RoleList, error) {
+	policyList, err := m.policyRegistry.ListPolicies(ctx, klabels.Everything(), klabels.Everything())
+	if err != nil {
+		return nil, err
+	}
+
+	roleList := &authorizationapi.RoleList{}
+
+	for _, policy := range policyList.Items {
+		for _, role := range policy.Roles {
+			roleList.Items = append(roleList.Items, role)
+		}
+	}
+
+	return roleList, nil
+}
+
+func (m *VirtualRegistry) GetRole(ctx kapi.Context, name string) (*authorizationapi.Role, error) {
+	policy, err := m.EnsurePolicy(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	role, exists := policy.Roles[name]
+	if !exists {
+		return nil, fmt.Errorf("role %v not found", name)
+	}
+
+	return &role, nil
+}
+
+func (m *VirtualRegistry) DeleteRole(ctx kapi.Context, name string) error {
+	policy, err := m.EnsurePolicy(ctx)
```

Comment:
Also feels weird for DELETE to create a policy... how do other registries behave if asked to delete something non-existent? Seems like we should treat the policy not existing the same as the role not existing within the policy.